examples/active_replication/flux_buffer.py

```diff
@@ -0,0 +1,54 @@
+
+class Buffer:
+    def __init__(self, n_dest):
+        self.n_dest = n_dest
+        self.queue = list()  # TODO(yika): add upper bound on buffer size
+
+    def put(self, data, sn, dest):
+        # Put input stream
+        if len(self.queue) > 0:
+            assert sn >= self.queue[-1][0]
+        self.queue.append((sn, data, dest))
+
+    def ack(self, sn, dest):
+        for i in range(len(self.queue)):
+            if self.queue[i][0] == sn and self.queue[i][2] == dest:
+                self.queue.pop(i)
+                return True
+        return False
+
+    def ack_all(self, dest):
+        i = 0
+        while i < len(self.queue):
+            if self.queue[i][2] == dest:
+                self.queue.pop(i)
+            else:
+                i += 1
+
+    def reset(self):
```

Comment:
I see this method is not used anywhere. Is it here for the recovery case? Could you please add a comment to explain what it does because it is not clear to me.